File: Domain/Enties/JobTitle.cs
namespace Domain.Enties
{
  using Domain.Account;
  using Domain.Entities;
  using Domain.Enums;
  using System;
  using System.Collections.Generic;
  using System.Linq;
  using System.Text;
  using System.Threading.Tasks;

  public class JobTitle : BaseEntity<int>
  {
    public string Title { get; set; }
    public string Description { get; set; }
    public Seniority Seniority { get; set; }

    public int DepartmentId { get; set; }
    public Department Department { get; set; }
    // Navigation property to Employee
    public ICollection<ApplicationUser> Users { get; set; }
  }
}

File: frontend/src/app/models/jobtitle.ts
import { BaseEntity } from "./Constant/baseEntity";

export interface JobTitle extends BaseEntity<number> {
  jobTitleId: number;
  title: string;
  departmentId: number;
  departmentName: string;
  description: string;
  seniority: 'Junior' | 'Mid' | 'Senior' | 'Lead';
}